src/services/RebalanceHLPService.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: BUSL-1.1
+// This code is made available under the terms and conditions of the Business Source License 1.1 (BUSL-1.1).
+// The act of publishing this code is driven by the aim to promote transparency and facilitate its utilization for educational purposes.
+
+pragma solidity 0.8.18;
+
+import { OwnableUpgradeable } from "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+// lib
+import { SafeERC20Upgradeable } from "@openzeppelin-upgradeable/contracts/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { IERC20Upgradeable } from "@openzeppelin-upgradeable/contracts/token/ERC20/IERC20Upgradeable.sol";
+
+// interface
+import { IGmxRewardRouterV2 } from "@hmx/interfaces/gmx/IGmxRewardRouterV2.sol";
+import { IGmxGlpManager } from "@hmx/interfaces/gmx/IGmxGlpManager.sol";
+import { ICalculator } from "@hmx/contracts/interfaces/ICalculator.sol";
+import { IVaultStorage } from "@hmx/storages/interfaces/IVaultStorage.sol";
+import { IRebalanceHLPService } from "@hmx/services/interfaces/IRebalanceHLPService.sol";
+
+contract RebalanceHLPService is OwnableUpgradeable, IRebalanceHLPService {
+  using SafeERC20Upgradeable for IERC20Upgradeable;
+
+  IERC20Upgradeable public sglp;
+  IVaultStorage public vaultStorage;
+  IGmxRewardRouterV2 public rewardRouter;
+  IGmxGlpManager public glpManager;
+
+  mapping(address => bool) public whitelistExecutors;
+
+  event LogSetWhitelistExecutor(address indexed _account, bool _active);
+
+  modifier onlyWhitelisted() {
+    // if not whitelist
+    if (!whitelistExecutors[msg.sender]) {
+      revert RebalanceHLPService_OnlyWhitelisted();
+    }
+    _;
+  }
```

Comment:
Let's use the whitelist from ConfigStorage instead.
See here https://github.com/HMXOrg/v2-evm/blob/main/src/services/TradeService.sol#L208.